block/header-and-data-separation-adr.md

```diff
@@ -0,0 +1,61 @@
+# Header and Data Separation ADR
+
+## Abstract
+
+The separation of header and data structures in Rollkit unlocks expanding the sequencing scheme beyond centralized sequencing and unlocks the use of a decentralized sequencer mode. This means that the creation of list of the transactions can be done by another network as well while rollup nodes still produce headers after executing that list of transactions. This overall change is akin to the proposer-builder separation in the Ethereum protocol, where the Rollkit header producer acts as the proposer, and the sequencer, which produces a list of transactions, acts as the builder.
+
+## Protocol/Component Description
+
+Before, Rollkit only supported the use of a centralized sequencer that was responsible for creating a list of rollup tx by reaping its mempool, executing them to produce a header, and putting them together in a rollup block. Rollkit headers and data were encapsulated within a single block structure. The block struct looked like this:
+
+```go
+// Block defines the structure of Rollkit block.
+type Block struct {
+	SignedHeader SignedHeader
+	Data         Data
+}
+```
+
+The separation of header and data into distinct structures allows them to be processed independently. The `SignedHeader` struct now focuses on the header information, while the `Data` struct handles transaction data separately. This separation is particularly beneficial in unlocking based rollups, where users submit transactions directly to the Data Availability layer which acts as the entity responsible for creating the list of transactions.
+
+This change also affects how rollup full nodes sync. Previously, rollup full nodes would apply the transactions from the `Block` struct and verify that the `header` in `SignedHeader` matched their locally produced header. Now, with the separation, full nodes obtain the transaction data separately (via the DA layer directly in based sequencer mode, or via p2p gossip/DA layer in centralized sequencer mode) and verify it against the header signed by the header producer once they have both components. This ensures that the data integrity and consistency are maintained across the network.
```

Comment:
can they get the header from DA as well?

Reply:
Yes full nodes can receive the header signed by the header producer from either p2p gossip or/and DA layer. If they got it from the p2p layer, they should wait to see the same header on the DA layer before marking the block as finalized in their view.